### ISSUE TITLE
LTR feature name

### ISSUE THREAD

**Issue by lintool** [castorini/anserini#1495]
I noticed some inconsistencies in feature naming... Before we push out a release, we should resolve.

A few questions: `ltr.feature` and `ltr.feature.base` was mostly legacy... do we still need sub-packages?

Strictly speaking, Java conventions want `Bm25` as opposed to `BM25`, but I'm fine with the latter. However, it also means we should have `IDFStat` and `TFIDFStat` right? And `TFIDFStat` should just be `TFIDF`? Or else you should have `BM25Stat` to be consistent?

Point is, let's go through the feature names and make sure we're happy with them?

**Comment by lintool** [comment]
Maybe the poolers should be directly under `ltr` and not `ltr.feature`? They're really not features?

**Comment by lintool** [comment]
Also, you have something like:

```
    @Override
    public String getName() {
        return String.format("%s_%s_BM25_HMean_k1_%.2f_b_%.2f_%s",field, qfield, k1, b, collectFun.getName());
    }
```

I would recommend using something like `this.getClass().getName()` so that if you rename the class, `getName` stays in sync.

**Comment by stephaniewhoo** [comment]
> I noticed some inconsistencies in feature naming... Before we push out a release, we should resolve.
> 
> A few questions: `ltr.feature` and `ltr.feature.base` was mostly legacy... do we still need sub-packages?
> 
> Strictly speaking, Java conventions want `Bm25` as opposed to `BM25`, but I'm fine with the latter. However, it also means we should have `IDFStat` and `TFIDFStat` right? And `TFIDFStat` should just be `TFIDF`? Or else you should have `BM25Stat` to be consistent?
> 
> Point is, let's go through the feature names and make sure we're happy with them?

we want to use `TFIDFStat` because we apply diff poolers to TFIDF. We name everything we apply different poolers with -Stat?

**Comment by stephaniewhoo** [comment]
let me know if you are ok with names [here](https://github.com/castorini/anserini/pull/1496). When we settle down names in anserini, I will change them in pyserini as well

**Comment by lintool** [comment]
I'm running regressions right now based on recent code additions - both LTR and the new bglinking tests. Will turn my attention to this next.

**Comment by lintool** [comment]
Closed by #1496
